Yosemite/Yosemite/Stores/TaxStore.swift

```diff
@@ -132,13 +149,51 @@ private extension TaxClassStore {
             storageTaxClass.update(with: readOnlyTaxClass)
         }
     }
-
 }
 
+// MARK: - Storage: TaxRate
+//
+private extension TaxStore {
+
+    /// Updates (OR Inserts) the specified ReadOnly TaxClass Entities *in a background thread*. onCompletion will be called
+    /// on the main thread!
+    ///
+    func upsertStoredTaxRatesInBackground(readOnlyTaxRates: [Networking.TaxRate], siteID: Int64, onCompletion: @escaping () -> Void) {
+        let derivedStorage = sharedDerivedStorage
+        derivedStorage.perform {
+            self.upsertStoredTaxRates(readOnlyTaxRates: readOnlyTaxRates, siteID: siteID, in: derivedStorage)
+        }
```

Comment:
Are we creating a potential retain cycle here? Since the method is marked as `@escaping` and `derivedStorage` is initialized lazily with the store, and then the store won't be deallocated through the app lifecycle.

```suggestion
        derivedStorage.perform { [weak self] in
            guard let self = self else { return }
            self.upsertStoredTaxRates(readOnlyTaxRates: readOnlyTaxRates, siteID: siteID, in: derivedStorage)
        }
```

Reply:
This is an interesting case, in most of the cases we don't weakify self when `derivedStorage.perform` in stores. It could be because the stores aren't deallocated during the app cycle (or perhaps they are when the user logs out?). In any case, I weakified it in bd62dbf1547ec0448f2d4dee5a74e7e4be0abaec just in case. Thanks for the suggestion!